vcd/datasource_vcd_vapp.go

```diff
@@ -0,0 +1,71 @@
+package vcd
+
+import "github.com/hashicorp/terraform/helper/schema"
+
+func datasourceVcdVApp() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceVcdVAppRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A unique name for the vApp",
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional description of the vApp",
+			},
+			"metadata": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Key value map of metadata to assign to this vApp. Key and value can be any string.",
+			},
+			"href": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"power_on": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
```

Comment:
"Default" probably not needed

Reply:
Yep, thanks. And also most of the `Optional` ones should be `Computed` instead. Fixing